### DISCUSSION TITLE
test results output file isnt created

### DISCUSSION THREAD

**Discussion by matan-yadgar** [xunit/xunit#3025] [Question] [answered]
Im running a parent process that prepares a test env (download stuff etc) and then triggers xUnit project (elaboration [here](https://stackoverflow.com/questions/78885392/dotnet-tests-file-generated-only-when-parent-process-exists)). Im waiting for some time (using WaitForExitAsync) as timeout. when the tests are finished (success/fail) within the timeout - the file is generated. However - when the timeout is exceeded - no file is generated. I tried to:
1) add the "--blame-hang-timeout <TIMESPAN>" flag. the file is generated but is empty. no .dmp files are found
2) terminate gracefully with SIGTERM
Am I missing something? How do I make sure that the file is created when the child process times out?
Thanks

**Comment by bradwilson** [comment]
I'm assuming (based on your command line switch example) that you're using `dotnet test`, in which case this is a question for [VSTest](https://github.com/microsoft/vstest/), as they are the ones responsible for generating the TRX report.

**Reply by matan-yadgar** [reply]
Im actually using the junit logger. this is an example of the command im running:

dotnet test C:\Windows\Temp\MyTests.dll --logger:"junit;LogFileName=C:\Windows\Temp\My_Folder\TestResults\test_res.xml" --verbosity normal --filter "test_name=my-test" --environment DATA_NEEDED_FOR_TEST=some-data --environment BRANCH_NAME=some-name

is this in your hands or some junit maintainers?

thanks!

**Reply by bradwilson** [reply, marked as answer]
Definitely not us, and probably not JUnit. If it shipped "in the box", then it's VSTest; if you had add some plugin to make it work, then the plugin author... but they may also point back to VSTest being the culprit.
